pkg/prometheus/statefulset.go

```diff
@@ -552,6 +552,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		},
 	}
 
+	promVolumeMounts = append(promVolumeMounts, p.Spec.VolumeMounts...)
```

Comment:
I am guessing this won't interfere with the merge patch we do?

Reply:
It should be fine because the container merge patch happens later.